src/core/Akka.Remote/RemoteDeploymentWatcher.cs

```diff
@@ -21,30 +16,31 @@ namespace Akka.Remote
     /// Responsible for cleaning up child references of remote deployed actors when remote node
     /// goes down (crash, network failure), i.e. triggered by Akka.Actor.Terminated.AddressTerminated
     /// </summary>
-    internal class RemoteDeploymentWatcher : ActorBase, IRequiresMessageQueue<IUnboundedMessageQueueSemantics>
+    internal class RemoteDeploymentWatcher : ReceiveActor, IRequiresMessageQueue<IUnboundedMessageQueueSemantics>
```

Comment:
ReceiveActor is a just C# DSL, and it could be change or eliminated when C# will support pattern matching. Do we really need this class inside system code?